File: Cpp/1636_Sort_Array_by_Increasing_Frequency.cpp
class Solution {
public:
    vector<int> frequencySort(vector<int>& nums) {
        vector<int> result;
        unordered_map<int, int> m;
        for(auto& num : nums){
            ++m[num];
        }
        
        vector<pair<int, int>> frequency;
        for(auto& it : m){
            frequency.push_back({it.second, it.first});
        }
        sort(frequency.begin(), frequency.end(),
             [](pair<int, int> a, pair<int, int> b) -> bool {
                 if(a.first == b.first) return a.second > b.second;
                 else return a.first < b.first;
        });
        
        for(auto& it : frequency){
            for(int i=0; i<it.first; ++i){
                result.push_back(it.second);
            }
        }
        
        return result;
    }
};

File: Kotlin/1636_Sort_Array_by_Increasing_Frequency.kt
class Solution {
    fun frequencySort(nums: IntArray): IntArray {
        val frequency = mutableMapOf<Int, Int>()
        
        for(num in nums) {
            frequency[num] = frequency.getOrDefault(num, 0) + 1
        }
        
        return nums.sortedWith(Comparator { a, b ->
            val aFrequency = frequency.getOrDefault(a, 0)
            val bFrequency = frequency.getOrDefault(b, 0)
            if (aFrequency == bFrequency) b.compareTo(a)
            else aFrequency.compareTo(bFrequency)
        }).toIntArray()
    }
}